### ISSUE TITLE
NodeProperty array<TYPO3\Media\Domain\Model\Asset> leads to Exception

### ISSUE THREAD

**Issue by knusperpixel** [neos/neos-development-collection#1384]
### Description

The `\TYPO3\Flow\Property\TypeConverter\StringConverter` is unable to convert properties of type `array<TYPO3\Media\Domain\Model\Asset>`.

When selecting a Node that has such a property in the reference editor NEOS breaks.

### Steps to Reproduce

1. Create a NodeType (a) with a property of type `array<TYPO3\Media\Domain\Model\Asset>`
2. Create another NodeType (b) with a property of type `reference`
3. Create a page of type (a) and add at least one asset
3. Create a page of type (b) and try to select node (a) in the references editor.

```
'Neos.Demo:Testtype1':
  superTypes:
    'TYPO3.Neos:Content': TRUE
  ui:
    label: 'Testnode1'
    group: 'general'
    icon: 'icon-picture'
    inspector:
      groups:
        'general':
          label: 'Settings'
  properties:
      myProperty:
        type: array<TYPO3\Media\Domain\Model\Asset>
        ui:
          reloadIfChanged: TRUE
          inspector:
            reloadIfChanged: TRUE
            group: 'general'

'Neos.Demo:Testtype2':
  superTypes:
    'TYPO3.Neos:Content': TRUE
  ui:
    label: 'Testnode2'
    group: 'general'
    icon: 'icon-picture'
    inspector:
      groups:
        'general':
          label: 'Settings'
  properties:
      myProperty:
        type: reference
        ui:
          reloadIfChanged: TRUE
          inspector:
            group: 'general'
            editorOptions:
              nodeTypes: ['TYPO3.Neos.NodeTypes:Page']
```

#### Expected behavior

Node (a) shows up in reference editor

#### Actual behavior

`PHP message: PHP Warning:  Uncaught TYPO3\Flow\Error\Exception: Catchable Fatal Error: Object of class TYPO3\Flow\Persistence\Doctrine\Proxies\__CG__\TYPO3\Media\Domain\Model\Image could not be converted to string in /var/www/my-domain.de/Data/Temporary/Production/Cache/Code/Flow_Object_Classes/TYPO3_Flow_Property_TypeConverter_StringConverter.php line 111 in /var/www/my-domain.de/Packages/Framework/TYPO3.Flow/Classes/TYPO3/Flow/Error/ErrorHandler.php:80`

This can be fixed by adding a `__toString` method to `\TYPO3\Media\Domain\Model\Asset` and `\TYPO3\Media\Domain\Model\Image`

### Affected Versions

Neos: 2.2.5
Flow: 3.2.5

[error.log.txt](https://github.com/neos/neos-development-collection/files/713799/error.log.txt)

**Comment by joachimmathes** [comment]
I will have a look at this issue.

**Comment by joachimmathes** [comment]
Hi @knusperpixel,

I changed the super type of `Testtype1` to `Page`, to make this example consistent:
```
'Neos.Demo:Testtype1':
  superTypes:
    'TYPO3.Neos.NodeTypes:Page': TRUE
  ui:
    label: 'Testnode1'
    group: 'general'
    icon: 'icon-picture'
    inspector:
      groups:
        'general':
          label: 'Settings'
  properties:
      myProperty:
        type: array<TYPO3\Media\Domain\Model\Asset>
        ui:
          reloadIfChanged: TRUE
          inspector:
            reloadIfChanged: TRUE
            group: 'general'

'Neos.Demo:Testtype2':
  superTypes:
    'TYPO3.Neos:Content': TRUE
  ui:
    label: 'Testnode2'
    group: 'general'
    icon: 'icon-picture'
    inspector:
      groups:
        'general':
          label: 'Settings'
  properties:
      myProperty:
        type: reference
        ui:
          reloadIfChanged: TRUE
          inspector:
            group: 'general'
            editorOptions:
              nodeTypes: ['TYPO3.Neos.NodeTypes:Page']
```

But I cannot reproduce the exception. Since NEOS version 2.2 is not supported anymore, I tested version 2.3 master and 3.0 master. 

Do you still run into this exception?

Best
Jo

**Comment by kitsunet** [comment]
This should be fixed with the new property conversion since 2.3
